model-datastructure/src/jvmMain/kotlin/org/modelix/model/KeyValueStoreWithStatistics.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model
+
+import java.util.concurrent.atomic.AtomicLong
+
+class KeyValueStoreWithStatistics(val store: IKeyValueStore) : IKeyValueStore {
```

Comment:
Delegation from the Kotlin language could be used to make implementation simpler.
https://kotlinlang.org/docs/delegation.html

(This is also interesting in many other places in our code.)